lib/devicetrust/authn/authn.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authn
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+
+	devicepb "github.com/gravitational/teleport/api/gen/proto/go/teleport/devicetrust/v1"
+	"github.com/gravitational/teleport/lib/devicetrust/native"
+)
+
+// vars below are used to swap native methods for fakes in tests.
+var (
```

Comment:
Nit: I can't say I'm a massive fan of this. I find that this style of faking/mocking can be hard to leverage when it comes to running tests in parallel (since only one override can be in place at once).

How would you feel about hanging the functions in this file off of a struct that held these dependencies instead?

Reply:
+1. Another option would be to add opts like `RunCeremony(ctx context.Context, devicesClient devicepb.DeviceTrustServiceClient, certs *devicepb.UserCertificates, opts ...CeremonyOpts)` to insert logic switches without changing the function signature of `RunCeremony` for non-test calls.

Reply:
Personally I can't say I mind the style much, nor do I think t.Parallel is a strong argument, but I hear ya. I have a similar testing strategy for lib/devicetrust/enroll, so I'll take a note to revisit both with the refactors you both suggested in mind.